### PR TITLE
JDK-8296661 : Typo Found In CSSParser.java

### PR DESCRIPTION
**Pull request by scientificware**
This is referenced in Java Bug Database as
- [JDK-8296661 : Typo Found In CSSParser.java](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8296661)

This is tracked in JBS as 
- [JDK-8296661 : Typo Found In CSSParser.java](https://bugs.openjdk.java.net/browse/JDK-8296661)

I suspect a typo in the documentation comments.

Designed from : [ScientificWare JDK-8296661 : Typo Found In CSSParser.java](https://github.com/scientificware/jdk/issues/18)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296661](https://bugs.openjdk.org/browse/JDK-8296661): Typo Found In CSSParser.java

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - no project role)
 * @SWinxy (no known github.com user name / role) ⚠️ Review applies to [4bfba96a](https://git.openjdk.org/jdk/pull/10975/files/4bfba96a43439111f2680ffe10dc0dcab5de9705)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [3a6eac4c](https://git.openjdk.org/jdk/pull/10975/files/3a6eac4c563c5ca9c6c90028f59b47840ebd41e8)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10975/head:pull/10975` \
`$ git checkout pull/10975`

Update a local copy of the PR: \
`$ git checkout pull/10975` \
`$ git pull https://git.openjdk.org/jdk pull/10975/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10975`

View PR using the GUI difftool: \
`$ git pr show -t 10975`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10975.diff">https://git.openjdk.org/jdk/pull/10975.diff</a>

</details>

### CODE REVIEW

**Review by andy-goryachev-oracle — APPROVED** [review]

**Review by SWinxy — APPROVED** [review]

**Comment by aivanov-jdk** [comment]
You have to update the subject of the PR to match the JBS title which was edited by someone to be more specific.

**Review by SWinxy — APPROVED** [review]

File: src/java.desktop/share/classes/javax/swing/text/html/CSSParser.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
  * <a href=http://www.w3.org/TR/REC-CSS1>http://www.w3.org/TR/REC-CSS1</a>.
  * If an error results in parsing, a RuntimeException will be thrown.
  * <p>
- * This will preserve case. If the callback wishes to treat certain poritions
+ * This will preserve case. If the callback wishes to treat certain portions
  * case insensitively (such as selectors), it should use toLowerCase, or
```
[the remark this answers is not in the document]
Comment (scientificware) [reply src/java.desktop/share/classes/javax/swing/text/html/CSSParser.java:53]: case-insensitively seems to be the most common usage in this repository.

**Review by aivanov-jdk — APPROVED** [review]

**Comment by scientificware** [comment]
@aivanov-jdk Done.

**Comment by scientificware** [comment]
Please, don't close this PR.

**Comment by SWinxy** [comment]
I think a Reviewer needs to re-review before it asks for sponsoring. @aivanov-jdk: re-approve changes?

File: src/java.desktop/share/classes/javax/swing/text/html/CSSParser.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1999, 2013, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1999, 2022, Oracle and/or its affiliates. All rights reserved.
```
Comment (andy-goryachev-oracle) [inline src/java.desktop/share/classes/javax/swing/text/html/CSSParser.java:2]: should it be 2023 now?

**Comment by andy-goryachev-oracle** [comment]
> Pre-submit tests - linux-x86 / test - Test (tier1) Failing after 67m — 1/9 failed

do we know why this test is failing?

**Comment by SWinxy** [comment]
> do we know why this test is failing?

Doesn't matter; it's javadoc changes only. Pre-submit tests randomly fail sometimes for no good reason.

**Review by prrace — APPROVED** [review]

**Comment by prrace** [comment]
> I think a Reviewer needs to re-review before it asks for sponsoring. @aivanov-jdk: re-approve changes?

No .. I think that the person who owns the PR needs to re-do the /integrate and that is what will make the sponsor label re-appear.

**Review by andy-goryachev-oracle — APPROVED** [review]
excellent

**Review by aivanov-jdk — APPROVED** [review]

**Comment by aivanov-jdk** [comment]
> @aivanov-jdk Done.

Sorry, @scientificware, I missed the update.

After making changes, you have to issue `integrate` command again.

In fact, I didn't mean to include [the changes I suggested](https://github.com/openjdk/jdk/pull/10975#issuecomment-1346460131) in this PR.

I still think there's room for improvement, however, it's not visible publicly. The list is inconsistent:

_“The delegate is notified of the following events:”_ — one would expect a list of events. Instead, each list item starts differently. If I may suggest, starting with each item with a callback method followed by the condition when it's called and additional details will present a more consistent documentation. The introduction sentence before the list may also be updated.

**Comment by scientificware** [comment]
@aivanov-jdk Thanks.

**Pull request closed without merging**